code_input_stream.h

```diff
@@ -0,0 +1,23 @@
+#ifndef MEGAZIP_CODE_INPUT_STREAM_H
```

Comment:
Why not use `#pragma once`? I know, it technically is not a part of the C++ standard, but it's supported by all major compilers and it looks cleaner
(also for consistency with the other files in this project)